app/controllers/rooms_controller.rb

```diff
@@ -0,0 +1,141 @@
+class RoomsController < ApplicationController
+  def show
+    p current_user.show_solutions.to_s
+    @setup = Setup.take
+    @week = @setup.weeks.find_by_number(params[:week_number])
+
+    @room = Room.find_by_id(params[:room_number])
+
+    @next_room = @room.next(current_user.id,@week.id)
+    @previous_room = @room.previous(current_user.id,@week.id)
+
+    @learning_objects = @room.learning_objects.order(id: :asc).distinct
+    @results = UserToLoRelation.get_results(current_user.id,@week.id)
+
+    @user = current_user
+  end
+
+  def eval
+    @setup = Setup.take
+    @room = Room.find(params[:room_number])
+    learning_objects = @room.learning_objects.all.distinct
+
+    weight_comment = 3
+
+    @value = @room.score_limit.to_f
+
+    @week=Week.find_by_id(@room.week_id)
+
+    if (@week.learning_objects.distinct.count % 10 == 0)
+      @all = (@week.learning_objects.distinct.count / 10).to_i
+    else
+      @all = (@week.learning_objects.distinct.count / 10).to_i + 1
+    end
+    @count = @week.rooms.where("user_id = ?", current_user.id).count
+
+    if @room.state.to_s != "used"
+      score1 = 0
+
+      learning_objects.each do |l|
+        feedbacks = l.feedbacks
+        order_to_comment = 0
+        order_to_comment_of_user = 0
+        feedbacks.each do |f|
```

Comment:
Technicky: Skryvanie nahradza vymazavanie zbytocnych komentarov, miesto DELETE z DB navzdy tam je, len je oflagovany, ze sa nema zobrazovat.

Ktore konkretne sa budu skryvat je otazka skor na profesorku. Aktualne su skryte vsetky minulorocne.  Napriklad diskusia medzi studentmi navzajom o danej otazke tam kludne moze zostat, takisto ak je tam nejaka chyba a administrator ju opravi, tak moze komentar nastavit ako vyrieseny a bude sa zobrazovat dalej (s textikom "VYRIESENE"), ale moze ho zaroven skryt, takze sa uz ani nebude zobrazovat (ale my mame info, ze bol vyrieseny).

Cize ak sa to ma zobrazovat studentom (teraz nestiham pozriet tuto metodu, co to vlastne robi), tak pouzivat iba viditelne. Ak sa z toho rata nejake hodnotenie, tak `nove`/`vyriesene`/`neriesene` a `zobrazovane`/`skryte` su dalsie indikatory komentara. (Napr. ak je vyrieseny a skryty bola tam chyba, ak je nerieseny a skryty, bola to blbost, atd.)